network/src/connectivity_manager/mod.rs

```diff
@@ -105,18 +111,29 @@ where
 {
     /// Creates a new instance of the [`ConnectivityManager`] actor.
     pub fn new(
+        self_peer_id: PeerId,
         eligible: Arc<RwLock<HashMap<PeerId, NetworkPublicKeys>>>,
+        seed_peers: HashMap<PeerId, Vec<Multiaddr>>,
         ticker: TTicker,
         peer_mgr_reqs_tx: PeerManagerRequestSender,
         control_notifs_rx: conn_status_channel::Receiver,
         requests_rx: channel::Receiver<ConnectivityRequest>,
         backoff_strategy: TBackoff,
         max_delay_ms: u64,
     ) -> Self {
+        // Ensure seed peers doesn't contain our own address (we want to avoid
+        // pointless self-dials).
+        let peer_addresses = seed_peers
+            .clone()
+            .into_iter()
+            .filter(|(peer_id, _)| *peer_id != self_peer_id)
+            .collect::<HashMap<PeerId, _>>();
```

Comment:
do you need the specialization here?

Reply:
I think we should be able to use `let peer_addresses: HashMap<_, _>>` =

Reply:
hmm I don't remember why I put the `PeerId` type hint... I think the compiler might have complained? the other question here is whether to put the type hints like:

```
let peer_addresses: HashMap<_, _> = /* iter */.collect();
```

or

```
let peer_addresses = /* iter */.collect::<HashMap<_, _>>();
```

I don't have any particularly strong opinion here